src/ToDoList.js

```diff
@@ -18,177 +16,142 @@ import {
   IconButton,
 } from "@mui/material";
 
-const containerStyle = {
-  display: "flex",
-  justifyContent: "center",
-  alignItems: "center",
-  width: "500px",
-  height: "600px",
-  backgroundColor: "lightGrey",
-  borderRadius: "30px",
-  margin: "0 auto",
-  marginTop: "30px",
-};
-
-const titleStyle = {
-  color: "black",
-  variant: "h1",
-  width: "300px",
-  height: "50px",
-  //borderRadius: "20px",
-  //backgroundColor: "lightBlue",
-  margin: "0 auto",
-  fontWeight: "bold",
-  //justifyContent: "center",
-  //alignItems: "center",
-};
-
-const inputContainerStyle = {
-  display: "flex",
-  justifyContent: "center",
-  alignItems: "center",
-  //height: "40px",
-};
-
-/*const taskItemStyle = {
-  width: "200px",
-  display: "flex",
-  display: "flex",
-  justifyContent: "center",
-  alignItems: "center",
-};*/
-
-const completedTaskStyle = {
-  textDecoration: "line-through",
-  textDecorationStyle: "solid",
-  textDecorationThickness: "2px",
-  color: "black",
-};
-
-/*const deleteButtonStyle = {
-  fontSize: "10px",
-  padding: "3px 3px",
-};*/
+const containerStyle = css`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+  gap: 30px;
+  h1 {
+    color: #29b6f6;
+    font-size: 50px;
+    font-weight: bold;
+    margin-top: 30px;
+  }
+`;
+
+const completedTaskStyle = css`
+  text-decoration: line-through solid 2px;
+  color: black;
+`;
+
+const listContainerStyle = css`
+  width: 100%;
+  max-height: 450px;
+  overflow-x: hidden;
+  overflow-y: auto;
+`;
 
 const ToDoList = () => {
-  useEffect(() => {
-    const storedTasks = localStorage.getItem("tasks");
-    if (storedTasks) {
-      setTasks(JSON.parse(storedTasks));
-    }
-  }, []);
-
-  const [tasks, setTasks] = useState(() => {
-    const storedTasks = localStorage.getItem("tasks");
+  const [newTaskValue, setNewTaskValue] = useState(() => {
```

Comment:
hmmm, something with the names of the variables is weird to me.
all the 3 variables sound like they store the same thing, and I think that there's a mix-up between all of the tasks and the new task variables including the local storage values.
```
newTaskValue
inputValue
taskState
```

also using a function in order to set an initial value for `useState` is not common, and I'm not sure if it's valid in React.

please try to do this:
const [tasks, setTasks] = useState(localStorage.getItem("tasks"));

Reply:
These are the names you suggested :) 
'tasks' and 'setTasks' are the original names, would you like me to revert to the original names?

Reply:
What you suggested does not work for me. I modified it to this: 
```
const storedTasks = localStorage.getItem("tasks");
  const initialTasks = storedTasks ? JSON.parse(storedTasks) : [];
  const [tasks, setTasks] = useState([...initialTasks]);
```
After reading about it I see that the difference between the two approaches is in when and how the initial state is computed. The older code allows for dynamic initialization based on the current value in localStorage, while the new code computes the initial state eagerly before the component is rendered.